kratos/solving_strategies/schemes/residual_based_pseudo_static_displacement_scheme.h

```diff
@@ -237,66 +229,64 @@ class ResidualBasedPseudoStaticDisplacementScheme
         const int disppos_z = it_node_begin->HasDofFor(DISPLACEMENT_Z) ? it_node_begin->GetDofPosition(DISPLACEMENT_Z) : -1;
         const int velpos_z = it_node_begin->HasDofFor(VELOCITY_Z) ? it_node_begin->GetDofPosition(VELOCITY_Z) : -1;
 
-        #pragma omp parallel for private(delta_displacement, predicted_x, predicted_y, predicted_z)
-        for(int i = 0;  i < num_nodes; ++i) {
-            auto it_node = it_node_begin + i;
+        block_for_each(rModelPart.Nodes(), delta_displacement, [&](Node<3>& rNode, array_1d<double,3>& rDeltaDisplacementTLS){
 
-            predicted_x = false;
-            predicted_y = false;
-            predicted_z = false;
+            bool predicted_x = false;
```

Comment:
```suggestion
            // TODO: This should be "private" and declared outside the loop once the utils are updated
            bool predicted_x = false;
```

Reply:
> @matekelemen I guess now you don't have to do this any more :)
> @ rubenzorrilla probably good to speak with me these days if you port some things, as my stundents and I are working on this too.
> => thats why I asked you to update the ConvergenceCritarias ;)
> 
> Also please use the methods I added in #8242 to reduce the code duplication

I don't really like the idea of doing things in the conditions before than doing so in the elements... note that some operations in the conditions might rely on the parent element ones, aside of the fact that using the entities utilities as they are right now would imply changing the behavior as the order is not the same.

We either change the order in the entities utilities or we leave it as it is right now.